diff --git a/source/vst/utility/alignedalloc.h b/source/vst/utility/alignedalloc.h
index 0ce85e7..d4c4e74 100644
--- a/source/vst/utility/alignedalloc.h
+++ b/source/vst/utility/alignedalloc.h
@@ -20,6 +20,7 @@
 #include "pluginterfaces/base/ftypes.h"
 
 #include <cstdlib>
+#include <new>
 
 #if __APPLE__
 #include <AvailabilityMacros.h>
@@ -55,6 +56,8 @@ inline void* aligned_alloc (size_t numBytes, uint32_t alignment)
 	posix_memalign (&data, alignment, numBytes);
 #elif defined(_MSC_VER)
 	data = _aligned_malloc (numBytes, alignment);
+#elif defined(__MINGW32__)
+	data = ::operator new (numBytes, std::align_val_t (alignment), std::nothrow);
 #else
 	data = std::aligned_alloc (alignment, numBytes);
 #endif
@@ -70,6 +73,8 @@ inline void aligned_free (void* addr, uint32_t alignment)
 	{
 #if defined(_MSC_VER)
 		_aligned_free (addr);
+#elif defined(__MINGW32__)
+		::operator delete (addr, std::align_val_t (alignment));
 #else
 		std::free (addr);
 #endif
